v1/remote/transport/transport.go

```diff
@@ -29,7 +29,7 @@ const (
 // New returns a new RoundTripper based on the provided RoundTripper that has been
 // setup to authenticate with the remote registry hosting "ref", in the capacity
 // laid out by the specified Scope.
-func New(ref name.Reference, auth authn.Authenticator, t http.RoundTripper, a Scope) (http.RoundTripper, error) {
+func New(ref name.Reference, auth authn.Authenticator, t http.RoundTripper, scopes []string) (http.RoundTripper, error) {
```

Comment:
I think this change would be a bit smaller if you used: `scopes ...string` instead

Reply:
Though I guess not as much as I'd hoped at first because we're also hoisting the `.Scope` calls.

Reply:
Making `scopes` variadic makes it harder to add more repeated args in the future, if we ever want to do that. I think I'd prefer to stick with `[]string`.

Reply:
I went back and forth about using an explicit slice vs. a variadic arg as well as whether we should keep the `Scope` type. I chose this implementation for the reason @ImJasonH suggests, and also because the data itself has changed from typed sentinel values to arbitrary data.

We should probably replace the `ref name.Reference` in `transport.New` with a `reg name.Registry`, given that only `ref.Context().Registry` is used. I think this would make the transport signature easier to understand. Any objections to that additional change?